File: app/src/main/java/com/sonyericsson/android/camera/debug/DebugMenuFragment.java
package com.sonyericsson.android.camera.debug;

import android.content.Context;
import android.content.SharedPreferences;
import android.content.SharedPreferences$OnSharedPreferenceChangeListener;
import android.os.Bundle;
import android.support.v14.preference.SwitchPreference;
import android.support.v7.preference.Preference;
import android.support.v7.preference.PreferenceFragmentCompat;
import android.support.v7.preference.PreferenceScreen;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;
import com.sonyericsson.android.camera.device.CameraInfo$CameraId;
import com.sonyericsson.android.camera.setting.SharedPreferencesAccessor;
import com.sonyericsson.android.camera.util.capability.PlatformCapability;
import kotlin.Metadata;
import kotlin.TypeCastException;
import kotlin.jvm.internal.Intrinsics;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/* JADX INFO: compiled from: DebugMenuFragment.kt */
/* JADX INFO: loaded from: classes.dex */
@Metadata(bv = {1, 0, 2}, d1 = {"\u0000R\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0002\b\b\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010\u000e\n\u0000\n\u0002\u0010\u000b\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0004\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0002\u0018\u00002\u00020\u00012\u00020\u0002B\u0005¢\u0006\u0002\u0010\u0003J\b\u0010\u0004\u001a\u00020\u0005H\u0002J\b\u0010\u0006\u001a\u00020\u0005H\u0002J\b\u0010\u0007\u001a\u00020\u0005H\u0002J\b\u0010\b\u001a\u00020\u0005H\u0002J\b\u0010\t\u001a\u00020\u0005H\u0002J\b\u0010\n\u001a\u00020\u0005H\u0002J\b\u0010\u000b\u001a\u00020\u0005H\u0002J\u0012\u0010\f\u001a\u00020\u00052\b\u0010\r\u001a\u0004\u0018\u00010\u000eH\u0016J\u0018\u0010\u000f\u001a\u00020\u00052\u0006\u0010\u0010\u001a\u00020\u00112\u0006\u0010\u0012\u001a\u00020\u0013H\u0016J\u001c\u0010\u0014\u001a\u00020\u00052\b\u0010\r\u001a\u0004\u0018\u00010\u000e2\b\u0010\u0015\u001a\u0004\u0018\u00010\u0016H\u0016J\u0010\u0010\u0017\u001a\u00020\u00182\u0006\u0010\u0019\u001a\u00020\u001aH\u0016J\b\u0010\u001b\u001a\u00020\u0005H\u0016J\b\u0010\u001c\u001a\u00020\u0005H\u0016J\u0018\u0010\u001d\u001a\u00020\u00052\u0006\u0010\u001e\u001a\u00020\u001f2\u0006\u0010 \u001a\u00020\u0016H\u0016J\u0010\u0010!\u001a\u00020\u00052\u0006\u0010\"\u001a\u00020#H\u0002J\b\u0010$\u001a\u00020\u0005H\u0002¨\u0006%"}, d2 = {"Lcom/sonyericsson/android/camera/debug/DebugMenuFragment;", "Landroid/support/v7/preference/PreferenceFragmentCompat;", "Landroid/content/SharedPreferences$OnSharedPreferenceChangeListener;", "()V", "clearFingerPrint", "", "initialize", "initializeAutoPowerOff", "initializeEmulateSideTouch", "initializeForceSettingMigration", "initializeLowPowerMode", "initializePredictiveCaptureOn", "onActivityCreated", "savedInstanceState", "Landroid/os/Bundle;", "onCreateOptionsMenu", "menu", "Landroid/view/Menu;", "inflater", "Landroid/view/MenuInflater;", "onCreatePreferences", "rootKey", "", "onOptionsItemSelected", "", "item", "Landroid/view/MenuItem;", "onPause", "onResume", "onSharedPreferenceChanged", "sharedPreferences", "Landroid/content/SharedPreferences;", "key", "reset", "context", "Landroid/content/Context;", "showKillProcessToast", "SemcCameraUI_release"}, k = 1, mv = {1, 1, 11})
public final class DebugMenuFragment extends PreferenceFragmentCompat implements SharedPreferences$OnSharedPreferenceChangeListener {
    public static final /* synthetic */ void access$clearFingerPrint(DebugMenuFragment debugMenuFragment) {
        debugMenuFragment.clearFingerPrint();
    }

    @Override // android.support.v7.preference.PreferenceFragmentCompat
    public void onCreatePreferences(@Nullable Bundle savedInstanceState, @Nullable String rootKey) {
        getPreferenceManager().setSharedPreferencesName("com.sonyericsson.android.camera.shared_preferences_debug");
        setPreferencesFromResource(2131886082, rootKey);
        initialize();
    }

    @Override // android.support.v4.app.Fragment
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        setHasOptionsMenu(true);
    }

    @Override // android.support.v4.app.Fragment
    public void onCreateOptionsMenu(@NotNull Menu menu, @NotNull MenuInflater inflater) {
        Intrinsics.checkParameterIsNotNull(menu, "menu");
        Intrinsics.checkParameterIsNotNull(inflater, "inflater");
        super.onCreateOptionsMenu(menu, inflater);
        inflater.inflate(2131558400, menu);
    }

    @Override // android.support.v4.app.Fragment
    public boolean onOptionsItemSelected(@NotNull MenuItem item) {
        Intrinsics.checkParameterIsNotNull(item, "item");
        Context context = getContext();
        if (!(context instanceof Context) || item.getItemId() != 2131296379) {
            return true;
        }
        reset(context);
        return true;
    }

    @Override // android.support.v4.app.Fragment
    public void onResume() {
        super.onResume();
        PreferenceScreen preferenceScreen = getPreferenceScreen();
        Intrinsics.checkExpressionValueIsNotNull(preferenceScreen, "preferenceScreen");
        SharedPreferences sharedPreferences = preferenceScreen.getSharedPreferences();
        if (sharedPreferences != null) {
            sharedPreferences.registerOnSharedPreferenceChangeListener(this);
        }
    }

    @Override // android.support.v4.app.Fragment
    public void onPause() {
        super.onPause();
        PreferenceScreen preferenceScreen = getPreferenceScreen();
        Intrinsics.checkExpressionValueIsNotNull(preferenceScreen, "preferenceScreen");
        SharedPreferences sharedPreferences = preferenceScreen.getSharedPreferences();
        if (sharedPreferences != null) {
            sharedPreferences.unregisterOnSharedPreferenceChangeListener(this);
        }
    }

    @Override // android.content.SharedPreferences$OnSharedPreferenceChangeListener
    public void onSharedPreferenceChanged(@NotNull SharedPreferences sharedPreferences, @NotNull String key) {
        Intrinsics.checkParameterIsNotNull(sharedPreferences, "sharedPreferences");
        Intrinsics.checkParameterIsNotNull(key, "key");
        int iHashCode = key.hashCode();
        if (iHashCode == -51439766) {
            if (key.equals("KEY_DEBUG_ALWAYS_PREDICTIVE_CAPTURE")) {
                Preference preferenceFindPreference = findPreference(key);
                if (preferenceFindPreference == null) {
                    throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
                }
                ((SwitchPreference) preferenceFindPreference).setChecked(sharedPreferences.getBoolean(key, false));
                return;
            }
            return;
        }
        if (iHashCode == 1006054472) {
            if (key.equals("KEY_DEBUG_DISABLE_AUTO_POWER_OFF")) {
                Preference preferenceFindPreference2 = findPreference(key);
                if (preferenceFindPreference2 == null) {
                    throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
                }
                ((SwitchPreference) preferenceFindPreference2).setChecked(sharedPreferences.getBoolean(key, false));
                return;
            }
            return;
        }
        if (iHashCode == 1901671088) {
            if (key.equals("KEY_DEBUG_EMULATE_SIDETOUCH")) {
                Preference preferenceFindPreference3 = findPreference(key);
                if (preferenceFindPreference3 == null) {
                    throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
                }
                ((SwitchPreference) preferenceFindPreference3).setChecked(sharedPreferences.getBoolean(key, false));
                return;
            }
            return;
        }
        if (iHashCode == 2012347019 && key.equals("KEY_DEBUG_DISABLE_LOW_POWER_MODE")) {
            Preference preferenceFindPreference4 = findPreference(key);
            if (preferenceFindPreference4 == null) {
                throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
            }
            ((SwitchPreference) preferenceFindPreference4).setChecked(sharedPreferences.getBoolean(key, false));
            showKillProcessToast();
        }
    }

    private final void initialize() {
        initializeAutoPowerOff();
        initializeLowPowerMode();
        initializeForceSettingMigration();
        initializePredictiveCaptureOn();
        initializeEmulateSideTouch();
    }

    private final void initializeAutoPowerOff() {
        Preference preferenceFindPreference = findPreference("KEY_DEBUG_DISABLE_AUTO_POWER_OFF");
        if (preferenceFindPreference == null) {
            throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
        }
        PreferenceScreen preferenceScreen = getPreferenceScreen();
        Intrinsics.checkExpressionValueIsNotNull(preferenceScreen, "preferenceScreen");
        ((SwitchPreference) preferenceFindPreference).setChecked(preferenceScreen.getSharedPreferences().getBoolean("KEY_DEBUG_DISABLE_AUTO_POWER_OFF", false));
    }

    private final void initializeLowPowerMode() {
        Preference preferenceFindPreference = findPreference("KEY_DEBUG_DISABLE_LOW_POWER_MODE");
        if (preferenceFindPreference == null) {
            throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
        }
        PreferenceScreen preferenceScreen = getPreferenceScreen();
        Intrinsics.checkExpressionValueIsNotNull(preferenceScreen, "preferenceScreen");
        ((SwitchPreference) preferenceFindPreference).setChecked(preferenceScreen.getSharedPreferences().getBoolean("KEY_DEBUG_DISABLE_LOW_POWER_MODE", false));
    }

    private final void initializeForceSettingMigration() {
        Preference settingMigration = findPreference("KEY_DEBUG_FORCE_MIGRATE_SETTINGS");
        Intrinsics.checkExpressionValueIsNotNull(settingMigration, "settingMigration");
        settingMigration.setOnPreferenceClickListener(new DebugMenuFragment$initializeForceSettingMigration$1(this));
    }

    private final void initializePredictiveCaptureOn() {
        Preference preferenceFindPreference = findPreference("KEY_DEBUG_ALWAYS_PREDICTIVE_CAPTURE");
        if (preferenceFindPreference == null) {
            throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
        }
        PreferenceScreen preferenceScreen = getPreferenceScreen();
        Intrinsics.checkExpressionValueIsNotNull(preferenceScreen, "preferenceScreen");
        ((SwitchPreference) preferenceFindPreference).setChecked(preferenceScreen.getSharedPreferences().getBoolean("KEY_DEBUG_ALWAYS_PREDICTIVE_CAPTURE", false));
    }

    private final void initializeEmulateSideTouch() {
        Preference preferenceFindPreference = findPreference("KEY_DEBUG_EMULATE_SIDETOUCH");
        if (preferenceFindPreference == null) {
            throw new TypeCastException("null cannot be cast to non-null type android.support.v14.preference.SwitchPreference");
        }
        PreferenceScreen preferenceScreen = getPreferenceScreen();
        Intrinsics.checkExpressionValueIsNotNull(preferenceScreen, "preferenceScreen");
        ((SwitchPreference) preferenceFindPreference).setChecked(preferenceScreen.getSharedPreferences().getBoolean("KEY_DEBUG_EMULATE_SIDETOUCH", false));
    }

    private final void reset(Context context) {
        DebugParameterUtils.INSTANCE.reset(context);
        initialize();
    }

    private final void showKillProcessToast() {
        Toast.makeText(getContext(), 2131690314, 0).show();
    }

    private final void clearFingerPrint() {
        for (CameraInfo$CameraId cameraInfo$CameraId : CameraInfo$CameraId.values()) {
            SharedPreferences sharedPreferences = SharedPreferencesAccessor.getSharedPreferences(getContext(), PlatformCapability.getFileNameForCameraCapability(cameraInfo$CameraId), 0);
            if (sharedPreferences == null) {
                return;
            }
            sharedPreferences.edit().remove("android.os.Build.FINGERPRINT").commit();
        }
        Toast.makeText(getContext(), 2131690313, 0).show();
    }
}

File: app/src/main/kotlin/com/sonyericsson/android/camera/debug/DebugMenuFragment.kt
package com.sonyericsson.android.camera.debug

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.support.v14.preference.SwitchPreference
import android.support.v7.preference.Preference
import android.support.v7.preference.PreferenceFragmentCompat
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.widget.Toast
import com.sonyericsson.android.camera.R
import com.sonyericsson.android.camera.device.CameraInfo
import com.sonyericsson.android.camera.setting.SharedPreferencesAccessor
import com.sonyericsson.android.camera.util.capability.PlatformCapability
@Suppress(names = ["DEPRECATION"])












class DebugMenuFragment : PreferenceFragmentCompat(), SharedPreferences.OnSharedPreferenceChangeListener {



    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        preferenceManager.setSharedPreferencesName("com.sonyericsson.android.camera.shared_preferences_debug")
        setPreferencesFromResource(R.xml.debug_preferences, rootKey)
        initialize()
    }


    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        setHasOptionsMenu(true)
    }


    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.debug_fragment_actionbar, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val context = context
        if (context is Context) {
            if (item.itemId == R.id.debug_fragment_actionbar_reset) {
                reset(context)
            }
        }
        return true
    }




    override fun onResume() {
        super.onResume()
        preferenceScreen.sharedPreferences?.registerOnSharedPreferenceChangeListener(this)
    }


    override fun onPause() {
        super.onPause()
        preferenceScreen.sharedPreferences?.unregisterOnSharedPreferenceChangeListener(this)
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String?) {
        when (key) {
            "KEY_DEBUG_DISABLE_AUTO_POWER_OFF" -> {
                val pref = findPreference(key) as SwitchPreference
                pref.setChecked(sharedPreferences.getBoolean(key, false))
            }
            "KEY_DEBUG_DISABLE_LOW_POWER_MODE" -> {
                val pref = findPreference(key) as SwitchPreference
                pref.setChecked(sharedPreferences.getBoolean(key, false))
                showKillProcessToast()
            }
            "KEY_DEBUG_ALWAYS_PREDICTIVE_CAPTURE" -> {
                val pref = findPreference(key) as SwitchPreference
                pref.setChecked(sharedPreferences.getBoolean(key, false))
            }
            "KEY_DEBUG_EMULATE_SIDETOUCH" -> {
                val pref = findPreference(key) as SwitchPreference
                pref.setChecked(sharedPreferences.getBoolean(key, false))
            }
        }
    }



    private fun initialize() {
        initializeAutoPowerOff()
        initializeLowPowerMode()
        initializeForceSettingMigration()
        initializePredictiveCaptureOn()
        initializeEmulateSideTouch()
    }


    private fun initializeAutoPowerOff() {
        val pref = findPreference("KEY_DEBUG_DISABLE_AUTO_POWER_OFF") as SwitchPreference

        val value = preferenceScreen.sharedPreferences
                .getBoolean("KEY_DEBUG_DISABLE_AUTO_POWER_OFF", false)
        pref.setChecked(value)
    }


    private fun initializeLowPowerMode() {
        val pref = findPreference("KEY_DEBUG_DISABLE_LOW_POWER_MODE") as SwitchPreference

        val value = preferenceScreen.sharedPreferences
                .getBoolean("KEY_DEBUG_DISABLE_LOW_POWER_MODE", false)
        pref.setChecked(value)
    }


    private fun initializeForceSettingMigration() {
        val settingMigration = findPreference("KEY_DEBUG_FORCE_MIGRATE_SETTINGS")

        settingMigration.setOnPreferenceClickListener {
            clearFingerPrint()
            true
        }
    }


    private fun initializePredictiveCaptureOn() {
        val pref = findPreference("KEY_DEBUG_ALWAYS_PREDICTIVE_CAPTURE") as SwitchPreference

        val value = preferenceScreen.sharedPreferences
                .getBoolean("KEY_DEBUG_ALWAYS_PREDICTIVE_CAPTURE", false)
        pref.setChecked(value)
    }


    private fun initializeEmulateSideTouch() {
        val pref = findPreference("KEY_DEBUG_EMULATE_SIDETOUCH") as SwitchPreference

        val value = preferenceScreen.sharedPreferences
                .getBoolean("KEY_DEBUG_EMULATE_SIDETOUCH", false)
        pref.setChecked(value)
    }

    private fun reset(context: Context) {
        DebugParameterUtils.reset(context)
        initialize()
    }


    private fun showKillProcessToast() {
        Toast.makeText(context, R.string.debug_preference_toast_kill_process, Toast.LENGTH_SHORT)
            .show()
    }


    private fun clearFingerPrint() {
        for (cameraId in CameraInfo.CameraId.values()) {
            val fileName = PlatformCapability.getFileNameForCameraCapability(cameraId)
            val sp = SharedPreferencesAccessor.getSharedPreferences(context, fileName, 0)
            if (sp == null) {
                return
            }
            sp.edit().remove("android.os.Build.FINGERPRINT").commit()
        }
        Toast.makeText(context, R.string.debug_preference_toast_execute_settings_migration, Toast.LENGTH_SHORT)
            .show()
    }
}
